routing/router.go

```diff
@@ -1369,19 +1386,8 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 			r.quit,
 		)
 		if err != nil {
-			// If we fail validation of the UTXO here, then we'll
-			// mark the channel as a zombie as otherwise, we may
-			// continue to continue to request it. We use the "zero
-			// key" for both node pubkeys so this edge can't be
-			// resurrected.
-			var zeroKey [33]byte
-			zErr := r.cfg.Graph.MarkEdgeZombie(
-				msg.ChannelID, zeroKey, zeroKey,
-			)
-			if zErr != nil {
-				return fmt.Errorf("unable to mark spent "+
-					"chan(id=%v) as a zombie: %w", msg.ChannelID,
-					zErr)
+			if err := r.addZombieEdge(msg.ChannelID); err != nil {
```

Comment:
Looks like we should only do this if we get a utxo not found error or something along those lines? Same applies for `GetBlockHash` and `GetBlock` in `fetchFundingTx`. This would prevent marking a valid edge as a non-resurrectable zombie if the RPC connection closes for whatever reason.

Reply:
Ah yeh I dug into this a bit, but then forgot that we return a specific error on the `lnd` side if the UTXO is indeed unspent, fixed!